apps/tx-builder/src/components/Dashboard.tsx

```diff
@@ -99,6 +74,47 @@ const Dashboard = () => {
     }
   }, [sdk.txs, transactions]);
 
+  // Load contract from address or ABI
+  useEffect(() => {
+    async function loadContract(addressOrAbi: string) {
+      setContract(null);
+      setLoadContractError('');
+
+      if (!addressOrAbi || !services.web3 || !services.interfaceRepo) {
+        return;
+      }
+
+      try {
+        setIsABILoading(true);
+        const contract = await services.interfaceRepo.loadAbi(addressOrAbi);
+        setContract(contract);
+      } catch (e) {
+        setContract(null);
+        setLoadContractError('No ABI found for this address');
+        console.error(e);
+      }
+      setIsABILoading(false);
+    }
+
+    loadContract(addressOrAbi);
+  }, [addressOrAbi, services.interfaceRepo, services.web3]);
+
+  function getAddressFromDomain(name: string): Promise<string> {
```

Comment:
Mixed use of arrow and normal functions in this file too

Reply:
done!